src/groups/bmq/bmqp/bmqp_protocol.h

```diff
@@ -3565,7 +3565,7 @@ inline RdaInfo::RdaInfo(unsigned int internalRepresentation)
 {
     BSLS_ASSERT_SAFE(internalRepresentation <= k_MAX_INTERNAL_COUNTER_VALUE);
 
-    d_counter = internalRepresentation;
+    d_counter = static_cast<unsigned char>(internalRepresentation);
```

Comment:
k_MAX_INTERNAL_COUNTER_VALUE is defined as 255U, so the BSLS_ASSERT_SAFE should guarantee that we don't run into data lost here